server/src/main/java/org/elasticsearch/health/node/LocalHealthMonitor.java

```diff
@@ -112,42 +135,80 @@ private void maybeScheduleNextRun(TimeValue time) {
     }
 
     // Helper method that starts the monitoring without a delay.
-    private void maybeScheduleNow() {
+    // Visible for testing
+    void maybeScheduleNow() {
         maybeScheduleNextRun(TimeValue.ZERO);
     }
 
     @Override
     public void clusterChanged(ClusterChangedEvent event) {
-        if (prerequisitesFulfilled == false) {
-            prerequisitesFulfilled = event.state().nodesIfRecovered().getMinNodeVersion().onOrAfter(Version.V_8_5_0)
-                && HealthMetadata.getFromClusterState(event.state()) != null;
-            maybeScheduleNow();
+        DiscoveryNode current = HealthNode.findHealthNode(event.state());
```

Comment:
nit: shall we name this `currentHealthNode`

Reply:
Yes, no objection.